bundles/automation/org.eclipse.smarthome.automation.event.test/src/test/groovy/org/eclipse/smarthome/automation/event/RuleEventTest.groovy

```diff
@@ -0,0 +1,170 @@
+        def condition1Config = [operator:"=", itemName:"myPresenceItem2", state:"ON"]
+        def condition2Config = [operator:"=", itemName:"myMotionItem2", state:"ON"]
+        def actionConfig = [itemName:"myLampItem2", command:"ON"]
+        def triggers = [
+            new Trigger("ItemStateChangeTrigger2", "ItemStateChangeTrigger", triggerConfig)
+        ]
+        def conditions = [
+            new Condition("ItemStateCondition3", "ItemStateCondition", condition1Config, null),
+            new Condition("ItemStateCondition4", "ItemStateCondition", condition2Config, null)
+        ]
+        def actions = [
+            new Action("ItemPostCommandAction2", "ItemPostCommandAction", actionConfig, null)
+        ]
+
+        def rule = new Rule("myRule21",triggers, conditions, actions, null, null)
+        rule.name="RuleEventTestingRule"
+
+        logger.info("Rule created: "+rule.getUID())
+
+        def ruleRegistry = getService(RuleRegistry)
+        ruleRegistry.add(rule)
+        ruleRegistry.setEnabled(rule.UID, true)
+
+        //TEST RULE
+
+        def EventPublisher eventPublisher = getService(EventPublisher)
+        def ItemRegistry itemRegistry = getService(ItemRegistry)
+        SwitchItem myMotionItem = itemRegistry.getItem("myMotionItem2")
+        Command commandObj = TypeParser.parseCommand(myMotionItem.getAcceptedCommandTypes(), "ON")
+        eventPublisher.post(ItemEventFactory.createCommandEvent("myPresenceItem2", commandObj))
+        //		eventPublisher.post(ItemEventFactory.createStateEvent("myPresenceItem2", OnOffType.ON))
+
+        Event itemEvent = null
+
+        def itemEventHandler = [
+            receive: {  Event e ->
+                logger.info("Event: " + e.topic)
+                if (e.topic.contains("myLampItem2")){
+                    itemEvent=e
+                }
+            },
+
+            getSubscribedEventTypes: {
+                Sets.newHashSet(ItemUpdatedEvent.TYPE, ItemStateEvent.TYPE)
+            },
+
+            getEventFilter:{ null }
+
+        ] as EventSubscriber
+
+        registerService(itemEventHandler)
+        commandObj = TypeParser.parseCommand(itemRegistry.getItem("myMotionItem2").getAcceptedCommandTypes(),"ON")
+        eventPublisher.post(ItemEventFactory.createCommandEvent("myMotionItem2", commandObj))
+        //		eventPublisher.post(ItemEventFactory.createStateEvent("myMotionItem2", OnOffType.ON))
+        waitForAssert ({ assertThat itemEvent, is(notNullValue())} , 3000, 100)
+        assertThat itemEvent.topic, is(equalTo("smarthome/items/myLampItem2/state"))
+        assertThat (((ItemStateEvent)itemEvent).itemState, is(OnOffType.ON))
+        def myLampItem2 = itemRegistry.getItem("myLampItem2")
+        assertThat myLampItem2, is(notNullValue())
+        logger.info("myLampItem2 State: " + myLampItem2.state)
+        assertThat myLampItem2.state, is(OnOffType.ON)
+        assertThat ruleEvents.size(), is(not(0))
```

Comment:
If something is not working as expected, I would really like to have a test failing as long as this issues isn´t fixed. This can help us getting it working.